src/daft-physical-plan/src/ops/placeholder.rs

```diff
@@ -0,0 +1,50 @@
+use std::sync::Arc;
+
+use common_display::{tree::TreeDisplay, DisplayLevel};
+use daft_logical_plan::ClusteringSpec;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct PlaceholderScan {
```

Comment:
Instead of naming this `PlaceholderScan`, we can call this something like `PreviousStageScan` and place the `stage_id` of the stage we expect to go inside of here. Then during the next iteration we can enforce the `stage_id` is what we expect